source/slang/decl-defs.h

```diff
@@ -90,21 +90,30 @@ SIMPLE_SYNTAX_CLASS(ClassDecl, AggTypeDecl)
 // An interface which other types can conform to
 SIMPLE_SYNTAX_CLASS(InterfaceDecl, AggTypeDecl)
 
+ABSTRACT_SYNTAX_CLASS(TypeConstraintDecl, Decl)
+    RAW(
+    virtual TypeExp& getSup() = 0;
+    )
+END_SYNTAX_CLASS()
+
 // A kind of pseudo-member that represents an explicit
 // or implicit inheritance relationship.
 //
-SYNTAX_CLASS(InheritanceDecl, Decl)
-    // The type expression as written
+SYNTAX_CLASS(InheritanceDecl, TypeConstraintDecl)
```

Comment:
This is where it seems a little odd for the base class to be called `TypeConstraintDecl`, since I don't think of an inheritance clause as expressing a constraint.

I almost wonder if what we want is to call the base class `SubtypeRelationshipDecl`.

Anyway, this is a "bike shed" sort of issue, and I'm fine with this name for now.